python/graphstorm/dataloading/sampler.py

```diff
@@ -389,3 +392,299 @@ def sample_blocks(self, g, seed_nodes, exclude_eids=None):
+
+class DistributedFileSampler(FileSamplerInterface):
+    r"""Distributed File Sampler. Samples file from the data path.
+    Each rank only has access to a partition of all the data shards.
+
+    Parameters:
+    ---------
+    dataset_path : str
+        path to the data files.
+        Mutually exclusive with :attr:`files`.
+    shuffle : bool
+        Set to ``True`` to have the files reshuffled
+        at every epoch. Shuffling is performed on the files of the local partition.
+    infinite : bool
+        Set to ``True`` to have the files sampled infinitely.
+    local_rank : int
+        Local rank ID.
+    world_size : int
+        Number of all trainers.
+    is_train : bool
+        Set to ``True`` if it's training set.
+    """
+    def __init__(
+        self, 
+        dataset_path=None, 
+        shuffle=False, 
+        infinite=False, 
+        local_rank=-1, 
+        world_size=1, 
+        is_train=True,
+    ):
+        super().__init__(dataset_path)
+
+        # Initialize distributed ranks
+        if local_rank == -1:
+            self.world_size = 1
+            self.local_rank = 0
+        else:
+            self.local_rank = local_rank
+            self.world_size = world_size
+
+        # Assign a slice window of file index to each worker.
+        # The slice window of each worker is specified
+        # by self.global_start and self.global_end
+        if self.world_size > self.num_files:
+            # If num of workers is greater than num of files,
+            # the slice windows are same across all workers,
+            # which covers all files.
+            self.remainder = self.world_size % self.num_files
+            self.global_start = 0
+            self.global_end = self.num_files
+            self.part_len = self.global_end
+        else:
+            # If num of workers is smaller than num of files,
+            # the slice windows are different for each worker.
+            # In the case where the files cannot be evenly distributed,
+            # the remainder will be assigned to one or multiple workers evenly.
+            part_len = self.num_files // self.world_size
+            self.remainder = self.num_files % self.world_size
+            self.global_start = part_len * self.local_rank + min(self.local_rank, self.remainder)
+            self.global_end = self.global_start + part_len + (self.local_rank < self.remainder)
+            self.part_len = self.global_end - self.global_start
+
+        if not is_train:
+            self.part_len = self.num_files
+        file_indices = list(range(self.part_len))
+        if shuffle and is_train:
+            sampler = RandomShuffleFileSampler(file_indices=file_indices)
+        else:
+            sampler = SequentialFileSampler(file_indices=file_indices, is_train=is_train)
+        self.sampler = sampler
+        self.sampler_iter = iter(sampler)
+
+        self.infinite = infinite
+        self.shuffle = shuffle
+
+
+    def get_file(self, offset):
```

Comment:
Can you write a unitest for this function?

Reply:
added unit test.

Reply:
Can you point me to the unit test, i.e., test_sampler_get_file.
Basically, you need to mimic different use cases like world_size > num_files and world_size <= num_files and different offset values.